public/js/annotations.js

```diff
@@ -0,0 +1,255 @@
+function Annotate(params) {
+  if (!(this instanceof Annotate)) {
+    // the constructor was called without "new".
+    return new Annotate(params);;
+  }
+  params = typeof params !== 'undefined' ? params : {};
+
+  this.tool      = null;
+  this.context   = null;
+  this.callbacks = null;
+
+  // I'd like this to be styled, but this is enough for MVP
+  this.lineWidth          = params.lineWidth           || 5;
+  this.lineColor          = params.lineColor           || "#df4b26";
+  this.fillColor          = params.fillColor           || "#F49638";
+  this.highlightRadius    = params.highlightRadius     || 20;
+  this.highlightPeriod    = params.highlightPeriod     || 500;
+  this.highlightFillColor = params.highlightFillColor  || 'rgba(245, 213, 213, 0.5)';
+  this.highlightLineColor = params.highlightLineColor  || '#cc0000';
+  this.zoom               = params.zoom                || 1; // Can you just die already?
```

Comment:
Yes, works properly in Firefox. (If it's not zooming, then it's just a couple `1 * 1` operations)